src/libostree/ostree-bootloader-syslinux.c

```diff
@@ -119,61 +106,49 @@ _ostree_bootloader_syslinux_write_config (OstreeBootloader          *bootloader,
                                      GCancellable          *cancellable,
                                      GError               **error)
 {
-  gboolean ret = FALSE;
   OstreeBootloaderSyslinux *self = OSTREE_BOOTLOADER_SYSLINUX (bootloader);
-  g_autoptr(GFile) new_config_path = NULL;
-  g_autofree char *config_contents = NULL;
-  g_autofree char *new_config_contents = NULL;
-  g_autoptr(GPtrArray) new_lines = NULL;
-  g_autoptr(GPtrArray) tmp_lines = NULL;
-  g_autofree char *kernel_arg = NULL;
-  gboolean saw_default = FALSE;
-  gboolean regenerate_default = FALSE;
-  gboolean parsing_label = FALSE;
-  char **lines = NULL;
-  char **iter;
-  guint i;
 
-  new_config_path = ot_gfile_resolve_path_printf (self->sysroot->path, "boot/loader.%d/syslinux.cfg",
-                                                  bootversion);
+  g_autoptr(GFile) new_config_path =
+    ot_gfile_resolve_path_printf (self->sysroot->path, "boot/loader.%d/syslinux.cfg", bootversion);
 
   /* This should follow the symbolic link to the current bootversion. */
-  config_contents = glnx_file_get_contents_utf8_at (AT_FDCWD, gs_file_get_path_cached (self->config_path), NULL,
-                                                    cancellable, error);
+  g_autofree char *config_contents =
+    glnx_file_get_contents_utf8_at (AT_FDCWD, gs_file_get_path_cached (self->config_path), NULL,
+                                    cancellable, error);
   if (!config_contents)
-    goto out;
+    return FALSE;
 
-  lines = g_strsplit (config_contents, "\n", -1);
-  new_lines = g_ptr_array_new_with_free_func (g_free);
-  tmp_lines = g_ptr_array_new_with_free_func (g_free);
-  
+  g_auto(GStrv) lines = g_strsplit (config_contents, "\n", -1);
+  g_autoptr(GPtrArray) new_lines = g_ptr_array_new_with_free_func (g_free);
+  g_autoptr(GPtrArray) tmp_lines = g_ptr_array_new_with_free_func (g_free);
+
+  g_autofree char *kernel_arg = NULL;
+  gboolean saw_default = FALSE;
+  gboolean regenerate_default = FALSE;
+  gboolean parsing_label = FALSE;
   /* Note special iteration condition here; we want to also loop one
    * more time at the end where line = NULL to ensure we finish off
    * processing the last LABEL.
    */
-  iter = lines;
-  while (TRUE)
+  for (char **iter = lines; iter; iter++)
     {
-      char *line = *iter;
+      /* Take ownership of the string element */
+      g_autofree char *line = g_steal_pointer (iter);
```

Comment:
Hmm, there's a subtle issue here (which looks like it existed even before). `g_strfreev` stops at the first `NULL` it finds. If we break out of this loop earlier than expected, the remaining strings will never be freed, right? It seems like it's simpler to just `g_strdup` in that case?

Reply:
Yeah, you're right.  Updated, also did some more re-indent while there ⬇️